cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/distributed/remote/RemoteExecutionJobMain.java

```diff
@@ -80,17 +85,32 @@ public static void main(String[] args) throws Exception {
    * @throws Exception if failed to the the job
    */
   private void doMain(String[] args) throws Exception {
-    if (args.length < 1) {
-      throw new IllegalArgumentException("Missing runId from the first argument");
+    Map<String, String> arguments = RuntimeArguments.fromPosixArray(args);
+    if (!arguments.containsKey(RUN_ID)) {
+      throw new IllegalArgumentException("Missing argument " + RUN_ID);
+    }
+
+    if (UserGroupInformation.isSecurityEnabled()) {
```

Comment:
Curious what sets this property, is this coming from the cluster environment?

Reply:
Yes. It is coming from the Hadoop configuration on the cluster.